### PR TITLE
Implemented Filecoin.ChainGetParentReceipts API

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:

- Updated test suite to identify identical error responses from `forest` and `lotus` as `Valid`
- Added support for `Filecoin.ChainGetParentReceipts` API
```
| RPC Method                           | Forest | Lotus |
|--------------------------------------|--------|-------|
| Filecoin.ChainGetParentReceipts (58) | Valid  | Valid |
```

## Reference issue to close (if applicable)

As part of https://github.com/ChainSafe/forest/issues/3639

## Change checklist

- [X] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation,
- [X] I have added tests that prove my fix is effective or that my feature works (if possible),
- [X] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by LesnyRumcajs — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]

**Review by lemmih — DISMISSED** [review]
Let's hold off on merging this until all the tests pass. If the tests are passing, please update the PR description to reflect this.

File: src/rpc_api/data_types.rs [hunk 1]
```diff
@@ -399,6 +399,19 @@ impl HasLotusJson for PendingBeneficiaryChange {
         }
     }
 }
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub struct ApiReceipt {
```
Comment (LesnyRumcajs) [inline src/rpc_api/data_types.rs:404]: Not sure if it's the practice all of you do with RPC methods, but it'd be great to document the fields we return.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/rpc/chain_api.rs [hunk 2]
```diff
@@ -58,6 +59,43 @@ pub(in crate::rpc) async fn chain_get_parent_message<DB: Blockstore>(
...
+    if block_header.epoch() == 0 {
+        Ok(LotusJson(vec![]))
+    } else {
+        let mut receipts = Vec::new();
+        let amt =
+            Amt::<Receipt, _>::load(block_header.message_receipts(), store).map_err(|_| {
+                JsonRpcError::Full {
+                    code: 1,
+                    message: format!(
+                        "failed to root: ipld: could not find {}",
```
[the remark this answers is not in the document]
Comment (lemmih) [reply src/rpc/chain_api.rs:79]: Gotta get them 0-day root hacks in the code. :)

**Comment by LesnyRumcajs** [comment]
@sudo-shashank there's still one outstanding comment that'd be great to address

**Review by LesnyRumcajs — APPROVED** [review]

File: src/rpc/chain_api.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sudo-shashank) [reply src/rpc/chain_api.rs]: this is too match lotus error

**Comment by sudo-shashank** [comment]
> @sudo-shashank there's still one outstanding comment that'd be great to address

@LesnyRumcajs which one?

File: src/rpc/chain_api.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply src/rpc/chain_api.rs]: We need to match error messages as well?

[the remark this answers is not in the document]
Comment (sudo-shashank) [reply src/rpc/chain_api.rs]: this is too mainly pass test cases in test suite

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**